### ISSUE TITLE
Cannot create index on user roles or groups

### ISSUE THREAD

**Issue by morhi** [statamic/cms#9230]
### Bug description

I want to customize the `query` method in the `UserRepository` to filter the users displayed in the control panel based on the logged in users permissions. This does work with the `super` field for example, but not with `roles` or `groups`. It seems like the Stache tries to create an index for these properties, however under Stache index location there is already a directory called `roles` and `groups`:

```bash
ll storage/framework/cache/data/stache/indexes/users/             
total 40
-rw-r--r--  1 user  staff    31B Dec 20 10:48 _indexes
-rw-r--r--  1 user  staff   793B Dec 20 10:48 email
drwxr-xr-x  6 user  staff   192B Dec 20 10:48 groups
-rw-r--r--  1 user  staff   897B Dec 20 10:48 id
-rw-r--r--  1 user  staff   1.5K Dec 20 10:48 path
drwxr-xr-x  9 user  staff   288B Dec 20 10:48 roles
-rw-r--r--  1 user  staff   485B Dec 20 10:48 super
```

I am not sure, where the directories `roles` and `groups` are coming from, but it seems like Stache is creating an index for all roles and groups, which are stored within the users index, which then creates an issue, because these indexes have the same name as the properties on the users blueprint.

### How to reproduce

1. Add an index to the users store under `config/statamic/stache.php`:

```php
    [
        'class' => Stores\UsersStore::class,
        'directory' => base_path('users'),
        'indexes' => [
            'super',
            'email',
            'groups',
            'roles',
        ]
    ]
```

2. Run `php please stache:refresh`

The exception then will be:

```
   ErrorException 

  file_put_contents(storage/framework/cache/data/stache/indexes/users/groups): Failed to open stream: Is a directory

  at vendor/laravel/framework/src/Illuminate/Filesystem/Filesystem.php:205
    201▕      * @return int|bool
    202▕      */
    203▕     public function put($path, $contents, $lock = false)
    204▕     {
  ➜ 205▕         return file_put_contents($path, $contents, $lock ? LOCK_EX : 0);
    206▕     }
    207▕ 
    208▕     /**
    209▕      * Write the contents of a file, replacing it atomically if it already exists.

      +31 vendor frames 
  32  please:37
      Illuminate\Foundation\Console\Kernel::handle(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))

```

### Environment

```yaml
Environment
Laravel Version: 10.38.0
PHP Version: 8.1.24
Composer Version: 2.5.8
Environment: local
Debug Mode: ENABLED
URL: localhost:8000
Maintenance Mode: OFF

Cache
Config: NOT CACHED
Events: NOT CACHED
Routes: NOT CACHED
Views: CACHED

Statamic
Addons: 12
Antlers: runtime
Stache Watcher: Disabled
Static Caching: Disabled
Version: 4.42.0 PRO
```

### Installation

Fresh statamic/statamic site via CLI

### Antlers Parser

Runtime (default)

**Comment by ryanmitchell** [comment]
This is already a known issue -  duplicate of: https://github.com/statamic/cms/issues/2498, and a fix is awaiting merge: https://github.com/statamic/cms/issues/2498

**Comment by morhi** [comment]
> This is already a known issue - duplicate of: #2498, and a fix is awaiting merge: #6131

Thanks for linking that, I didn't find those issues. Sounds good, that a fix is on the way 🙏
